Automattic/wp-calypso
# Stats: remove feedback and opt out notices

Related to #80472 

## Proposed Changes

> - The opt-out notice is being shown unnecessarily to new Jetpack customers visiting the Odyssey Stats dashboard for the first time.
>- The opt-out notice has served its purpose of informing our customers that the older stats dashboard still exists.
> - We don’t have anyone monitoring the survey results as closely as we used to, so it’s no longer a viable place for customers to send us feedback.

Based on the above reasons, the PR proposes to remove Opt-out and Feedback notices.

As the `LegacyNotices` was intended for lower Jetpack versions, and we are removing all of its notices, the compatibility layer would not be needed anymore, the PR proposes to remove it and its dependencies all together.

The next step is to refactor the notices so that the logic is clear to show only one notice a time.


## Testing Instructions

#### Calypso 
* Open Calypso Live Branch
* Open a site without Paid Stats and upgrade notices not dismissed `/stats/day/:siteSlug`
* Ensure you could see the upgrade notice

#### Test Odyssey Stats Locally

* Build Odyssey Stats `STATS_PACKAGE_PATH=~/path/to/jetpack/projects/packages/stats-admin yarn dev`
* Build your local Jetpack if necessary `jetpack build plugins/jetpack`
* Open `/wp-admin/admin.php?page=stats`
* Ensure you could see the upgrade notice
* Purchase a Paid Stats subscription following the upgrade link
* Ensure you see the upgrade successful notice when returning to Stats Dashboard
* Refresh several time
* Ensure you do not see any other notices

## Pre-merge Checklist
